infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogEntryWriter.java

```diff
@@ -148,21 +147,14 @@ public long apply(LogReplicationEntryMsg msg) throws ReplicationWriterException
 
         // Skip entries that have already been processed
         if (msg.getMetadata().getTimestamp() <= lastMsgTs) {
-            log.warn("Ignore Log Entry. Received message with snapshot {} is smaller than lastMsgTs {}.",
-                    msg.getMetadata().getSnapshotTimestamp(), lastMsgTs);
+            log.warn("Ignore Log Entry. Received message with timestamp {} is smaller than lastMsgTs {}.",
```

Comment:
One question, in the case the ACK was missed by the active node. Let's assume it resends the exact same batch (that was already applied on standby but not acked), meaning that TS == lastMsgTs. We would continue to ignore the message and never come out, isn't it?

Reply:
Yes, We would ignore the msg, but still send an ACK for it with the "lastProcessedTs" metadata which is populated to what was last processed. So the log replication should move ahead even if standby receives a msg repeatedly in-order/out-of-order.